app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,110 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
+  before_action :load_and_authorize_posts, except: [:create, :reply]
+
+  def create
+    if generate_post
+      redirect_to create_success_path
+    else
+      redirect_to create_fail_path
+    end
+  end
+
+  def edit
```

Comment:
this is the abstract controller. There's no implementation for edit, no view for edit... so what's the point?